server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/MockMessageHandler.java

```diff
@@ -57,29 +55,25 @@
         @Override
         public void behave(T input) throws RejectException {
             Response response = behavior.getResponse();
-            if (response.isServerRejected()) {
-                throw new RejectException(response.getCode().getRawCode(), response.getMessage());
-            }
-            throw new NotImplementedException("Not rejecting commands in mock behaviours is not supported yet");
+            throw new RejectException(response.getCode().getRawCode(), response.getMessage());
```

Comment:
Take directly a `Response`

Reply:
RejectException belongs to the library we use; not our code... Cannot change that; sorry.